### DISCUSSION TITLE
[Node Operator Question] Huge resource usage difference of op-geth for read only ethereum node and third party provider node

### DISCUSSION THREAD

**Discussion by aytunc-tunay** [ethereum-optimism/developers#205] [General]
### Are you running the most up to date node software?

- [X] Yes

### Did you check the documentation?

- [X] Yes

### Did you check for duplicate questions?

- [X] Yes

### Issue Description

I'm running two different optimism node. The one on the left is connected to alchemy, the one on the right connected to our own besuteku read only ethereum node. Even though both of them have similar speed and block number (currently around 112 million) , their storage and ram usage so different, and I couldnt understand the reason behind. Can someone help me to understand this ?

Args for op-node connected to own read only ethereum node: 
            
'--l1=http://besu-service:8545/'
'--l1.rpckind=basic'
'--l1.trustrpc'

Args for op-node connected to alchemy node: 
            
'--l1=https://eth-mainnet.g.alchemy.com/'
'--l1.rpckind=alchemy'
'--l1.trustrpc'

![Screenshot 2024-01-22 at 13 44 32](https://github.com/ethereum-optimism/developers/assets/136096855/5da450d3-1ca0-4e67-99c1-c31c4dd42e0d)

### Protocol Description

op-geth: us-docker.pkg.dev/oplabs-tools-artifacts/images/op-geth:v1.101304.1
op-node: us-docker.pkg.dev/oplabs-tools-artifacts/images/op-node:6a9f126e2f0503b55bd949fcdb7a3d7607ac0c9d
network: mainnet

**Comment by opfocus** [comment]
Just out of curiosity, are their sync modes the same? Are they all achieve mode or full mode?

**Reply by trianglesphere** [reply]
Bumping this @aytunc-tunay. If this issue is still around, could you get the head block hash & number from both of these op-geth nodes?

**Comment by aytunc-tunay** [comment]
Both are using this - '--syncmode=full'

**Reply by opfocus** [reply]
Is there any node set with --gcmode archive?   Or the host of node running block explorer at the same time

**Reply by aytunc-tunay** [reply]
No, there is not.No, our Ethereum node is not set up in --gcmode archive. It is running Hyperledger Besu with --sync-mode=X_SNAP and --data-storage-format=BONSAI. We are not running a block explorer or any similar heavy services directly on this node

**Reply by opfocus** [reply]
Sorry, I don't know the reason.  ....
 I feel that the picture on the right is more reasonable.

**Comment by sbvegan** [comment]
I've asked around, initially we were thinking that because Alchemy might ve fetching all receipts individually so that might have been the cause of the data build up. But you're using the `--l1.rpckind=alchemy` so it should be optimized. 

Can you increase the SSD size on the Alchemy node? It looks like its reaching capacity, maybe thats why the memory is working so hard? It looks like its reaching a threshold on those spikes and then the garbage collector is freeing up space.

I'm kind of stumped on this one.
